hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/container/balancer/ContainerBalancer.java

```diff
@@ -348,17 +361,24 @@ private boolean initializeIteration() {
   }
 
   private IterationResult doIteration() {
+    // note that potential and selected targets are updated in the following
+    // loop
     List<DatanodeDetails> potentialTargets = getPotentialTargets();
     Set<DatanodeDetails> selectedTargets =
         new HashSet<>(potentialTargets.size());
     moveSelectionToFutureMap = new HashMap<>(unBalancedNodes.size());
 
     // match each overUtilized node with a target
     for (DatanodeUsageInfo datanodeUsageInfo : overUtilizedNodes) {
+      if (!balancerRunning || Thread.currentThread().isInterrupted()) {
```

Comment:
when reaching here, is there a case `balancerRunning` is true and meanwhile `Thread.currentThread().isInterrupted()` is true? we can only check `balancerRunning`, maybe that is enough

Reply:
Yes, I think checking only `balancerRunning` is enough. I'll change this.